src/main/scala/org/constellation/primitives/RandomTransactionManager.scala

```diff
@@ -33,83 +47,82 @@ object RandomTransactionManager {
-              dao.metricsManager ! IncrementMetric("signaturesPerformed")
-              dao.metricsManager ! IncrementMetric("randomTransactionsGenerated")
-              dao.metricsManager ! IncrementMetric("sentTransactions")
+            // TODO: Make deterministic buckets for tx hashes later to process based on node ids.
+            // this is super easy, just combine the hashes with ID hashes and take the max with BigInt
 
-              dao.threadSafeTXMemPool.put(tx)
-              /*            // TODO: Change to transport layer call
-      dao.peerManager ! APIBroadcast(
-        _.put(s"transaction/${tx.edge.signedObservationEdge.signatureBatch.hash}", tx),
-        peerSubset = Set(getRandomPeer._1)
-      )*/
-            }
-          }
+            def getRandomPeer: (Id, PeerData) = peerIds(Random.nextInt(peerIds.size))
 
-          if (
-            memPoolCount > dao.processingConfig.minCheckpointFormationThreshold &&
-              dao.generateRandomTX &&
-              dao.nodeState == NodeState.Ready &&
-              !dao.blockFormationInProgress
-          ) {
-
-            dao.blockFormationInProgress = true
-
-            val messages = dao.threadSafeMessageMemPool.pull(1).getOrElse(Seq())
-            futureTryWithTimeoutMetric(
-              EdgeProcessor.formCheckpoint(messages),
-              "formCheckpointFromRandomTXManager",
-              timeoutSeconds = dao.processingConfig.formCheckpointTimeout,
-              {
-                messages.foreach { m =>
-                  dao.threadSafeMessageMemPool.activeChannels(m.signedMessageData.data.channelId).release()
-                }
-                dao.blockFormationInProgress = false
-              }
-            )(dao.edgeExecutionContext, dao)
-          }
-          dao.metricsManager ! UpdateMetric("blockFormationInProgress", dao.blockFormationInProgress.toString)
+            val sendRequest = SendToAddress(getRandomPeer._1.address, Random.nextInt(1000).toLong + 1L, normalized = false)
+            val tx = createTransaction(dao.selfAddressStr, sendRequest.dst, sendRequest.amount, dao.keyPair, normalized = false)
+            dao.metrics.incrementMetric("signaturesPerformed")
+            dao.metrics.incrementMetric("randomTransactionsGenerated")
+            dao.metrics.incrementMetric("sentTransactions")
 
+            dao.threadSafeTXMemPool.put(tx)
+            /*            // TODO: Change to transport layer call
+    dao.peerManager ! APIBroadcast(
+      _.put(s"transaction/${tx.edge.signedObservationEdge.signatureBatch.hash}", tx),
+      peerSubset = Set(getRandomPeer._1)
+    )*/
+          }
         }
-      }
 
+        if (
+          memPoolCount > dao.processingConfig.minCheckpointFormationThreshold &&
+            dao.generateRandomTX &&
+            dao.nodeState == NodeState.Ready &&
+            !dao.blockFormationInProgress
+        ) {
+
+          dao.blockFormationInProgress = true
```

Comment:
I don't think so? I made the Periodic class await the result of the last future before scheduling another one. Either way, this could be a lot better I agree...